### PR TITLE
Add Pinch Gestures and improved other gesture recognizers

### LINKED ISSUE #3430: Scroll start for gestures should be configurable
I'm using a 7" touchscreen with a resolution of 800 x 480.
The value for `ScrollGestureRecognizer.ScrollStartDistance` is 30 pixels wich mean in my case about 6 millimeters. When i want to scroll i have to move my finger 6 mm before the scrolling is starting and the content first jump these 6 mm before it continue to scroll.

`ScrollGestureRecognizer.ScrollStartDistance` should be configurable or better be dependent of the screen resolution.

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Adds a pinch-zoom gesture recognizer, `PinchGestureRecognizer`, along with some changes to the other gesture recognizers.

**Changes**
1. `ScrollGestureRecognizer.ScrollStartDistance` is now a DIrectProperty
2. Removed redundant pointer event handlers in `PullGestureRecognizer`
3. Added a Gestures sample page demonstrating Pull and Pinch-Zoom Gestures

**Added API**

- `PinchGestureRecognizer`
- `PinchEventArgs`
    - `double Scale { get; }`
    - `Point ScaleOrigin { get; }`
- `PinchEndedEventArgs`

PinchGestureRecognizer does not do any direct manipulation of the attached visuals.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #3430

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027796-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs [hunk 1]
```diff
@@ -75,12 +94,9 @@ public void PointerPressed(PointerPressedEventArgs e)
                 _trackedRootPoint = e.GetPosition((Visual?)_target);
             }
         }
-
-        // Arbitrary chosen value, probably need to move that to platform settings or something
-        private const double ScrollStartDistance = 30;
         
         // Pixels per second speed that is considered to be the stop of inertial scroll
-        private const double InertialScrollSpeedEnd = 5;
+        private const double InertialScrollSpeedEnd = 0;
```
Comment (maxkatz6) [inline src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs:99]: Was it changed intentionally?

File: samples/ControlCatalog/Pages/GesturePage.cs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            if (control == null)
+            {
+                return;
+            }
+
+            var ball = control.FindLogicalDescendantOfType<Border>();
+
+            Vector3 defaultOffset = default;
+
+            CompositionVisual? ballCompositionVisual = null;
```
Comment (maxkatz6) [inline samples/ControlCatalog/Pages/GesturePage.cs:149]: Composition samples might be too complex for somebody, who just want to know how to use these gestures. But I am fine with that either way.

**Review by maxkatz6 — APPROVED** [review]

File: src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs (hunk 1 above)
Comment (emmauss) [reply src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs:99]: That was unintentional

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027823-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**